addon/services/key-manager.js

```diff
@@ -79,72 +81,61 @@ export default Ember.Service.extend({
     });
   },
 
-  handler(event) {
-    if (get(this, 'isDestroyed') || get(this, 'isDestroying')) {
-      return;
-    }
-    const {
-      data,
-      keyCode,
-    } = event;
+  findMatchingCombo(event) {
+    try {
+      const { data, keyCode } = event;
+      const { eventName = null } = data;
+
+      let combo = this._findComboByName(eventName);
+      if (!combo) { return; }
 
-    if (!modifierKeyCodes.includes(keyCode)) {
-      get(this, 'downKeys').addObject(keyCode);
+      const isNotOnInput = inputElements.every((e) => {
+        return !$(document.activeElement).is(e);
+      });
+
+      if (!get(combo, 'disableOnInput') || isNotOnInput) {
+        const callback = get(combo, 'callback') || function() {};
+        callback(event);
+      }
+    } finally {
+      this._clearDownKeys(event);
```

Comment:
We wrap the function in a `try-finally`, so that even if we `return` early in the function, we still clear the `downKeys` array.